python/lsst/ts/ess/labjack/base_labjack_data_client.py

```diff
@@ -164,14 +168,42 @@ def _blocking_connect(self) -> None:
         self.handle = ljm.openS(
             self.config.device_type, self.config.connection_type, identifier
         )
+        self._blocking_stop_data_stream()
 
     def _blocking_disconnect(self) -> None:
-        """Disconnect from the LabJack. A no-op if disconnected.
+        """Disconnect from the LabJack after stopping streaming.
+
+        A no-op if disconnected.
 
         Call in a thread to avoid blocking the event loop.
         """
         if self.handle is not None:
             try:
+                self._blocking_stop_data_stream()
                 ljm.close(self.handle)
             finally:
                 self.handle = None
+
+    def _blocking_stop_data_stream(self) -> None:
+        """Stop streaming data from the LabJack.
+
+        Call in a thread to avoid blocking the event loop.
+
+        This logs errors but should not raise an exception.
+        """
+        # LabJack ljm demo mode does not support streaming,
+        # but this call seems to work anyway.
+        try:
+            ljm.eStreamStop(self.handle)
+        except ljm.LJMError as e:
+            # Note: I would rather compare e.errorCode to
+            # ljm.errorcodes.STREAM_NOT_RUNNING
+            # but ljm seems to have a bug; the error code is 2620
```

Comment:
Might be worth adding what error code 2620 is.

Reply:
I have no idea, but I'll add that to the comment.